tests/models/test_hparams.py

```diff
@@ -554,3 +556,25 @@ def test_args(tmpdir):
     with pytest.raises(TypeError, match="__init__\(\) got an unexpected keyword argument 'test'"):
         SubClassVarArgs.load_from_checkpoint(raw_checkpoint_path)
 
+
+class UnsafeParamModel(BoringModel):
+
+    def __init__(self, my_path, any_param=123):
```

Comment:
Could you try to pass tensor and np.array to see if self.save_hyperparameters() works 👍

Reply:
it works, even `torch.nn.Module`